### PR TITLE
Move build scripts into its own folder; add unittests

### PR DESCRIPTION
**Pull request by queengooborg**
This PR moves the build scripts into its own folder separate from the `release` folder.  Additionally, this adds unittests to ensure that our transforms are functioning as expected.

### CODE REVIEW

File: scripts/build/index.ts [hunk 1]
```diff
@@ -110,18 +110,26 @@ export const addVersionLast = (feature: WalkOutput): void => {
 };
 
 /**
```
Comment (Elchi3) [inline scripts/build/index.ts:112]: Should we have `/* c8 ignore start */`  in line 112 instead of line 140?

File: scripts/build/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+              version_added: '10',
+              version_removed: '20',
+            },
+            firefox: [
+              {
+                version_added: '18',
+              },
+              {
+                version_added: '15',
+                version_removed: '16',
```
Comment (Elchi3) [inline scripts/build/index.test.ts:65]: I'd change this to something a bit more interesting. Here it should produce version_last 3.6 (and not 3, for example).

```suggestion
                version_added: '1',
                version_removed: '4',
```

File: scripts/build/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    for (const feature of walker) {
+      addVersionLast(feature);
+    }
+
+    assert.equal(data.feature.__compat.support.chrome.version_last, '19');
+    assert.equal(
+      data.feature.__compat.support.firefox[0].version_last,
+      undefined,
+    );
+    assert.equal(data.feature.__compat.support.firefox[1].version_last, '15');
```
Comment (Elchi3) [inline scripts/build/index.test.ts:83]:
```suggestion
    assert.equal(data.feature.__compat.support.firefox[1].version_last, '3.6');
    assert.equal(data.feature.__compat.support.safari.version_last, 'preview');
    assert.equal(data.feature.__compat.support.edge.version_last, true);
```

File: scripts/build/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            },
+            firefox: [
+              {
+                version_added: '18',
+              },
+              {
+                version_added: '15',
+                version_removed: '16',
+              },
+            ],
```
Comment (Elchi3) [inline scripts/build/index.test.ts:67]: I would add two more cases to test what happens with `true` and `preview`:

```suggestion
            ],
            safari: {
              version_added: '10',
              version_removed: 'preview',
              version_last: 'preview',
            },
            edge: {
              version_added: '12',
              version_removed: true,
              version_last: true,
            },
```

File: scripts/build/index.test.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    assert.equal(data.feature.__compat.support.chrome.version_last, '19');
+    assert.equal(
+      data.feature.__compat.support.firefox[0].version_last,
+      undefined,
+    );
+    assert.equal(data.feature.__compat.support.firefox[1].version_last, '15');
+  });
+});
```
Comment (Elchi3) [inline scripts/build/index.test.ts:85]: Might be good to test manifest creation as well (and completing test coverage for this file).

```suggestion
  it('createManifest', () => {
    const manifest = createManifest();
    assert.ok(manifest.main);
    assert.ok(manifest.exports);
    assert.ok(manifest.types);
    assert.ok(manifest.name);
    assert.ok(manifest.description);
    assert.ok(manifest.repository);
    assert.ok(manifest.keywords);
    assert.ok(manifest.author);
    assert.ok(manifest.license);
    assert.ok(manifest.bugs);
    assert.ok(manifest.homepage);
  });
});
```

File: scripts/build/index.test.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
+/* This file is a part of @mdn/browser-compat-data
+ * See LICENSE file for more information. */
+
+import assert from 'node:assert/strict';
+
+import { walk } from '../../utils/index.js';
+
+import { generateMeta, applyMirroring, addVersionLast } from './index.js';
```
Comment (Elchi3) [inline scripts/build/index.test.ts:8]: Lets test manifest creating as well.

```suggestion
import {
  generateMeta,
  applyMirroring,
  addVersionLast,
  createManifest,
} from './index.js';
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Thanks, this is great! I just have some proposals for more tests scenarios for `version_last` and to complete the code coverage.

File: scripts/build/index.ts (hunk 1 above)
Comment (queengooborg) [reply scripts/build/index.ts:112]: I'm not sure!  Do you think it would be good to test the data bundle creation?

Comment (Elchi3) [reply scripts/build/index.ts:112]: hm yes, I think you are right. I was just looking at the coverage report and saw these lines still uncovered. Do you want to write some tests for this?

Comment (queengooborg) [reply scripts/build/index.ts:112]: I'd be down, but would it be alright to bump those changes to the next PR instead?  (Currently unable to do so right now, happy to explain why in a DM!)

**Review by Elchi3 — APPROVED** [review]
Thank you! 👍

**Pull request merged**